### PR TITLE
columnCount() of null fixed

### PR DESCRIPTION
**Pull request by shopblocks**

### CODE REVIEW

**Comment by shopblocks** [comment]
@davidpersson We can't get the integration tests to run for mysql locally it skips them because of a missing 'test' connection.

**Comment by shopblocks** [comment]
Managed to figure it out. It had to be part of an actual application, and couldn't be ran independently. If you remove our fix, you receive the fatal error. With the fix, it passes.

File: tests/integration/data/source/database/adapter/MySqlTest.php [hunk 1]
```diff
@@ -353,6 +353,18 @@ public function testDefaultValues() {
...
+
+	/**
+	 * Verifies that setting options using a raw SQL string works, when
+	 * the operation returns no result.
+	 *
+	 * @link https://github.com/UnionOfRAD/lithium/issues/1210
+	 */
+	public function testRawOptionSettingWithNoResultResource() {
+		$expected = array();
+		$result = Galleries::connection()->read('SET SESSION group_concat_max_len = 100000;');
```
Comment (davidpersson) [inline tests/integration/data/source/database/adapter/MySqlTest.php]: As the query is actually executed, following tests might be affected by the change. Is there any kind of query that fulfills our test purposes but does not change the server/client behavior?

Comment (shopblocks) [inline tests/integration/data/source/database/adapter/MySqlTest.php]: Im afraid that I cannot say for certain whether there is, and I dont have time to check this. Perhaps we can set a session on something that doesn't actually matter? Or perhaps reset the session afterwards?

Comment (koushki) [inline tests/integration/data/source/database/adapter/MySqlTest.php]: I think this is similar `SET @temp = 1`, could you please test this one ?

Comment (shopblocks) [inline tests/integration/data/source/database/adapter/MySqlTest.php]: My Tests show that this works too! Thanks @koushki 

Commit made to change this.

**Comment by shopblocks** [comment]
@davidpersson Managed to get this to not change settings on the server

**Comment by davidpersson** [comment]
Good. Can you squash the commits together and force push? Then we should be ready to merge. Thanks!

**Comment by shopblocks** [comment]
@davidpersson Done.

**Comment by alexbowers** [comment]
@davidpersson Hey, any chance of a merge? Thanks.

**Comment by davidpersson** [comment]
This is currently on hold. We will merge this when release plans have been sorted out.

**Comment by davidpersson** [comment]
Fixed and merged in 191d29589911fba1517e163eb72bacdf2f948e5c

**Comment by alexbowers** [comment]
Thanks.

**Pull request closed without merging**